homeassistant/components/switchbot_cloud/sensor.py

```diff
@@ -0,0 +1,93 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitchbotCloudData
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+from .entity import SwitchBotCloudEntity
+
+SENSOR_TYPE_TEMPERATURE = "temperature"
+SENSOR_TYPE_HUMIDITY = "humidity"
+SENSOR_TYPE_BATTERY = "battery"
+
+METER_PLUS_SENSOR_DESCRIPTIONS = (
+    SensorEntityDescription(
+        key=SENSOR_TYPE_TEMPERATURE,
+        name="Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_TYPE_HUMIDITY,
+        name="Humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_TYPE_BATTERY,
+        name="Battery",
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+
+    async_add_entities(
+        SwitchBotCloudSensor(data.api, device, coordinator, description)
+        for device, coordinator in data.devices.sensors
+        for description in METER_PLUS_SENSOR_DESCRIPTIONS
+    )
+
+
+class SwitchBotCloudSensor(
+    SwitchBotCloudEntity, CoordinatorEntity[SwitchBotCoordinator], SensorEntity
+):
+    """Representation of a SwitchBot Cloud sensor entity."""
+
+    def __init__(
+        self,
+        api: SwitchBotAPI,
+        device: Device,
+        coordinator: SwitchBotCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize SwitchBot Cloud sensor entity."""
+        super().__init__(api, device, coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{device.device_id}_{description.key}"
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        if self.coordinator.data:
+            if self.entity_description.key == SENSOR_TYPE_TEMPERATURE:
+                self._attr_native_value = self.coordinator.data.get("temperature")
+            if self.entity_description.key == SENSOR_TYPE_HUMIDITY:
+                self._attr_native_value = self.coordinator.data.get("humidity")
+            if self.entity_description.key == SENSOR_TYPE_BATTERY:
+                self._attr_native_value = self.coordinator.data.get("battery")
+            self.async_write_ha_state()
```

Comment:
Either extend SensorEntityDescription to add a `value_fn` or do
```suggestion
        self._attr_native_value = self.coordinator.data.get(self.entity_description.key)
        self.async_write_ha_state()
```

I think because of the CoordinatorEntity, coordinator.data is always there